src/account.ts

```diff
@@ -757,6 +795,50 @@ export class ArgentAccount extends Account {
         return signatures;
     }
 
+    public override async estimateDeployAccountFee(): Promise<starknet.FeeEstimation> {
+        this.assertNotDeployed();
+        const hre = await import("hardhat");
+
+        const nonce = numericToHexString(0);
+        const maxFee = numericToHexString(0);
+
+        const constructorCalldata: string[] = [
+            ArgentAccount.IMPLEMENTATION_CLASS_HASH,
+            hash.getSelectorFromName("initialize"),
+            "2",
+            this.publicKey,
+            ArgentAccount.generateGuardianPublicKey(this.guardianPrivateKey)
+        ].map((val) => BigInt(val).toString());
+        const calldataHash = hash.computeHashOnElements([
+            ArgentAccount.PROXY_CLASS_HASH,
+            this.salt,
+            ...constructorCalldata
+        ]);
+        const msgHash = hash.computeHashOnElements([
+            TransactionHashPrefix.DEPLOY_ACCOUNT,
+            numericToHexString(QUERY_VERSION),
+            this.address,
+            0, // entrypoint selector is implied
+            calldataHash,
+            maxFee,
+            hre.starknet.networkConfig.starknetChainId,
+            nonce
+        ]);
+
+        const signature = this.getSignatures(msgHash);
+        const data = {
+            type: "DEPLOY_ACCOUNT",
+            class_hash: ArgentAccount.PROXY_CLASS_HASH,
+            constructor_calldata: constructorCalldata,
+            contract_address_salt: this.salt,
+            signature: bnToDecimalStringArray(signature || []),
+            version: numericToHexString(QUERY_VERSION),
+            nonce: numericToHexString(0)
```

Comment:
Why not use the defined `nonce` variable?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/304/commits/eac6546509a07710b6ef2776f4f66b6e4f8db708